# Hypothesis Testing - Assignment 3

Q.1 Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions?

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [2]:
data=pd.read_csv("C:\\Users\\lenovo\\Downloads\\BuyerRatio.csv")

In [3]:
data.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [4]:
males=pd.Series(data.iloc[0,1:])
males

East      50
West     142
North    131
South     70
Name: 0, dtype: object

In [5]:
females=pd.Series(data.iloc[1,1:])
females

East      435
West     1523
North    1356
South     750
Name: 1, dtype: object

In [6]:
p_value=stats.ttest_ind(males,females)

In [7]:
p_value

Ttest_indResult(statistic=-3.5844638979607404, pvalue=0.011580726207478622)

In [8]:
p_value[1]

0.011580726207478622

pvalue is less than 0.05 ,the buyer ratio of males and females are different.

Q.2 TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

In [9]:
import numpy as np
import pandas as pd
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [10]:
data1=pd.read_csv("C:\\Users\\lenovo\\Downloads\\Costomer+OrderForm.csv")

In [11]:
data1.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [12]:
data1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Phillippines  300 non-null    object
 1   Indonesia     300 non-null    object
 2   Malta         300 non-null    object
 3   India         300 non-null    object
dtypes: object(4)
memory usage: 9.5+ KB


In [13]:
data1.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [14]:
data1.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [15]:
data1.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [16]:
data1.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [17]:
data1.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [18]:
obs=np.array([[271,267,269,280],[29,33,31,20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

Ho: Independence of categorical variables (customer order forms defective % does not varies by centre)
Ha: Dependence of categorical variables (customer order forms defective % varies by centre).

In [19]:
chi2_contingency(obs)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

Q3. A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

In [20]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [21]:
data2=pd.read_csv("C:\\Users\\lenovo\\Downloads\\Cutlets.csv")

In [22]:
data2.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [23]:
data2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Unit A  35 non-null     float64
 1   Unit B  35 non-null     float64
dtypes: float64(2)
memory usage: 688.0 bytes


In [24]:
data.describe()

East         West        North       South
count    2.000000     2.000000     2.000000    2.000000
mean   242.500000   832.500000   743.500000  410.000000
std    272.236111   976.514465   866.205807  480.832611
min     50.000000   142.000000   131.000000   70.000000
25%    146.250000   487.250000   437.250000  240.000000
50%    242.500000   832.500000   743.500000  410.000000
75%    338.750000  1177.750000  1049.750000  580.000000
max    435.000000  1523.000000  1356.000000  750.000000

In [25]:
df=pd.Series(data2.iloc[:,0])

In [26]:
df

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [27]:
df1=pd.Series(data2.iloc[:,1])

In [28]:
df

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [29]:
p_value=stats.ttest_ind(df,df1)
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

In [30]:
p_value[1]

0.4722394724599501

Inference : As(p_value=0.472)>(α = 0.05); Accept null hypothesis.

Q.4 A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch. Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

In [31]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [32]:
data3=pd.read_csv("C:\\Users\\lenovo\\Downloads\\LabTAT.csv")

In [33]:
data3.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [34]:
p_value=stats.f_oneway(data3.iloc[:,0],data3.iloc[:,1],data3.iloc[:,2],data3.iloc[:,3])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [35]:
p_value[1]

2.1156708949992414e-57

Inference: As (p-value=0) < (α = 0.05); Reject Null Hypothesis.